api.proto

```diff
@@ -10,6 +10,13 @@ message CreateStreamRequest {
     string group             = 3; // Partitions NATS subject amongst group members
     int32  replicationFactor = 4; // Number of stream replicas
     int32  partitions        = 5; // Number of stream partitions
+	int64                    RetentionMaxBytes     =6; // Custom StreamConfig
+	int64                    RetentionMaxMessages  =7; // Custom StreamConfig
+	google.protobuf.Duration RetentionMaxAge       =8; // Custom StreamConfig  
+	google.protobuf.Duration CleanerInterval       =9; // Custom StreamConfig
+	int64                    SegmentMaxBytes       =10; // Custom StreamConfig
+	google.protobuf.Duration SegmentMaxAge         =11; // Custom StreamConfig
+	int64                    CompactMaxGoroutines  =12; // Custom StreamConfig
```

Comment:
Does this need a `CompactEnabled` `bool` for enabling compaction?

Reply:
yes it is, I was thinking about activate `compact.enabled` automatically whenever this setting is found. However after a while I think that the this should be left for the client to do it, which gives them more flexibility and less implicit behavior.